llama_cpp/llama.py

```diff
@@ -328,6 +333,10 @@ def _sample(
             else last_n_tokens_size
         )
         logits = self.eval_logits[-1]
+        for processor in logits_processors:
+            logits = processor(list(self.eval_tokens), logits)
+
+        self.eval_logits[-1] = logits
```

Comment:
Does logits_processor have to modify the logits, can it simply update the logits array here during the sampling?

Reply:
Yes the logits does have to modify the logits.